src/RecordComponent.cpp

```diff
@@ -176,6 +178,39 @@ RecordComponent::empty() const
     return *m_isEmpty;
 }
 
+void
+RecordComponent::serialize()
+{
+    // verify: read-only
+    if( IOHandler()->m_frontendAccess != Access::READ_ONLY )
+        throw std::runtime_error("serialize only works on read-only access");
+    //if( IOHandler()->m_backendAccess != Access::READ_ONLY )
+    // as well??
+
+    // verify: no MPI
+    auto & series = this->retrieveSeries();
+    auto const str_backend = series.backend(); // HDF5, ADIOS2, ...
+    if( str_backend.find( "MPI" ) != std::string::npos )
+        throw std::runtime_error("serialize does not work with MPI");
+
+    // information to construct a new serial read-only series
+    auto const str_path = IOHandler()->directory; // samples/git-samples/
+    auto const str_name = series.name();          // data%T
+    auto const str_ext = suffix(series.m_format); // .bp, .h5, .json, ...
+    auto const filepath = str_path + str_name + str_ext;
```

Comment:
@franzpoeschel I might add this logic as the first argument in `.mypath` logic with this PR, just to keep it all together. What do you think?

Reply:
So you want to add that information to the output of `.myPath()`?
If yes, I would suggest to make this a little struct
```cpp
struct MyPath
{
    std::string directory;
    std::string seriesName;
    std::string seriesExtension;
    std::string filePath() const;
    std::vector< std::string > openPMDGroup;
}
```
It's already error-prone enough to handle the `std::vector<std::string>` correctly, but at least somewhat justifiable since it's an openPMD path reconstructed by the `Attributable` backbone in openPMD. Also, this would keep the old functionality of `.myPath()` separate.